pkg/users/local.js

```diff
@@ -47,6 +47,22 @@ const sortGroups = groups => {
     });
 };
 
+export function get_next_available_id(already_used, is_groups, change) {
+    //  Handle also the case where logindef == null, i.e. the file does not exist.
+    //  While that's unusual, "empty /etc" is a goal, and it shouldn't crash the page.
+    cockpit.file("/etc/login.defs").read()
+            .then(logindef => {
```

Comment:
Counterintuitively, cockpit.file().read() returns `null` for nonexisting files, see [API docs](https://cockpit-project.org/guide/latest/cockpit-file.html). This means that (1) this code will crash if logindef is null, and (2) the error handler of the promise should do a console.error(), as it's *really* unexpected to fail.

I know this code was merely moved, but let's fix it while at it.

Reply:
This wasn't resolved at all, the [current commit](https://github.com/cockpit-project/cockpit/pull/18353/commits/e07ce0f830a938f7c898ab69da22780c41545740) is still the same. Curiously, the comment at the top is even aware of this behaviour, it's just not at all how it's implemented..

Reply:
I'll reopen a new thread, apparently this old one keeps getting ignored.